app/controllers/admin/claim_approvals_controller.rb

```diff
@@ -0,0 +1,9 @@
+class Admin::ClaimApprovalsController < Admin::BaseAdminController
+  before_action :ensure_service_operator
+
+  def create
+    @claim = Claim.find(params[:claim_id])
+    @claim.approve!(approved_by: admin_session.user_id)
```

Comment:
I feel like this code will potentially mean the application lies. i.e. if you call the action on a claim that cannot be approved, it will redirect you and tell you the "Claim has been approved successfully" even though it hasn't. Should we instead conditionally do the redirect only if the approval is successful and give a suitable message if isn't?